lang/py/src/avro/schema.py

```diff
@@ -405,6 +424,83 @@ def __eq__(self, that):
     return self.props == that.props
 
 #
+# Decimal Type
+#
+class DecimalSchema(LogicalSchema):
+  def __init__(self, decimal_type, precision, scale=0, size=0, fixed_schema=None):
+    if decimal_type != 'fixed' and decimal_type != 'bytes':
+      raise SchemaParseException('Logical type DECIMAL must be backed by fixed or bytes.')
+
+    if fixed_schema:
+      self._fullname = fixed_schema._fullname
+      self.fixed_schema_props = fixed_schema.props
+
+    # Call parent ctor
+    LogicalSchema.__init__(self, decimal_type, 'decimal')
+
+    self.set_prop('precision', precision)
+    self.set_prop('scale', scale)
+    if decimal_type == 'fixed':
+      self.props.update(self.fixed_schema_props)
+      if size <= 0:
+        raise SchemaParseException('Size of a fixed length should be greater than 0')
+      self.set_prop('size', size)
+
+    self.validate()
+
+  # read-only properties
+  size = property(lambda self: self.get_prop('size'))
+  name = property(lambda self: self.get_prop('name'))
+  namespace = property(lambda self: self.get_prop('namespace'))
+  fullname = property(lambda self: self._fullname)
+
+  def validate(self):
+    if not isinstance(self.get_prop('precision'), int):
+      raise SchemaParseException('Precision (Integer) is required for logical type DECIMAL.')
+
+    precision = self.get_prop('precision')
+    if precision <= 0:
+      raise SchemaParseException("Invalid DECIMAL precision %s. Must be positive."
+                                 % precision)
```

Comment:
I was just being consistent with the other places in the file where exceptions were thrown.